mm2src/mm2_main/src/lp_native_dex.rs

```diff
@@ -549,21 +487,76 @@ fn get_p2p_key(ctx: &MmArc, i_am_seed: bool) -> P2PResult<[u8; 32]> {
     Ok(p2p_key)
 }
 
-pub async fn init_p2p(ctx: MmArc) -> P2PResult<()> {
-    let i_am_seed = ctx.is_seed_node();
+fn p2p_precheck(ctx: &MmArc) -> P2PResult<()> {
+    let is_seed_node = ctx.is_seed_node();
+    let is_bootstrap_node = ctx.is_bootstrap_node();
+    let disable_p2p = ctx.disable_p2p();
+    let p2p_in_memory = ctx.p2p_in_memory();
     let netid = ctx.netid();
 
     if DEPRECATED_NETID_LIST.contains(&netid) {
         return MmError::err(P2PInitError::InvalidNetId(NetIdError::Deprecated { netid }));
     }
 
+    let seednodes = seednodes(ctx)?;
+
+    if is_bootstrap_node && !is_seed_node {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Bootstrap node must also be a seed node.".to_owned(),
+        });
+    }
+
+    if is_bootstrap_node && !seednodes.is_empty() {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Bootstrap node cannot have seed nodes to connect.".to_owned(),
+        });
+    }
+
+    if !is_bootstrap_node && seednodes.is_empty() && !disable_p2p {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Non-bootstrap node must have seed nodes configured to connect.".to_owned(),
+        });
+    }
+
+    if disable_p2p && !seednodes.is_empty() {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Cannot disable P2P while seed nodes are configured.".to_owned(),
+        });
+    }
+
+    if disable_p2p && p2p_in_memory {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Cannot disable P2P while using in-memory P2P mode.".to_owned(),
+        });
+    }
+
+    if disable_p2p && is_seed_node {
+        return MmError::err(P2PInitError::Precheck {
+            reason: "Seed nodes cannot disable P2P.".to_owned(),
+        });
+    }
```

Comment:
you can group checks related to `disable_p2p` together
```rust
     // P2P disable constraints
    if disable_p2p {
        if !seednodes.is_empty() {
            ...
        }
        if p2p_in_memory {
            ...
        }
        if is_seed_node {
            ...
        }
    }
```

Reply:
That's better yeah, thanks.